src/ar/cimaleek/src/eu/kanade/tachiyomi/animeextension/ar/cimaleek/Cimaleek.kt

```diff
@@ -0,0 +1,305 @@
+        val anime = SAnime.create()
+        anime.thumbnail_url = document.select("div.ani_detail-stage div.film-poster img").attr("src")
+        anime.title = document.select("div.anisc-more-info div.item:contains(الاسم) span:nth-child(3)").text()
+        anime.author = document.select("div.anisc-more-info div.item:contains(البلد) span:nth-child(3)").text()
+        anime.genre = document.select("div.anisc-detail div.item-list a").joinToString(", ") { it.text() }
+        anime.description = document.select("div.anisc-detail div.film-description div.text").text()
+        anime.status = if (document.select("div.anisc-detail div.item-list").text().contains("افلام")) SAnime.COMPLETED else SAnime.UNKNOWN
+        return anime
+    }
+
+    // ============================ Video Links =============================
+    override fun videoFromElement(element: Element): Video = throw UnsupportedOperationException()
+
+    override fun videoUrlParse(document: Document): String = throw UnsupportedOperationException()
+
+    override fun videoListSelector(): String = "div#servers-content div.server-item div"
+
+    override fun videoListParse(response: Response): List<Video> {
+        val document = response.asJsoup()
+        val script = document.selectFirst("script:containsData(dtAjax)")!!.data()
+        val version = script.substringAfter("ver\":\"").substringBefore("\"")
+        return document.select(videoListSelector()).parallelCatchingFlatMapBlocking {
+            extractVideos(it, version)
+        }
+    }
+
+    private fun extractVideos(element: Element, version: String): List<Video> {
+        fun generateRandomString(length: Int): String {
+            val characters = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789"
+            val result = StringBuilder(length)
+            for (i in 0 until length) {
+                val randomIndex = (Math.random() * characters.length).toInt()
+                result.append(characters[randomIndex])
+            }
+            return result.toString()
+        }
+        val videoList = mutableListOf<Video>()
+        val videoUrl = "$baseUrl/wp-json/lalaplayer/v2/".toHttpUrlOrNull()!!.newBuilder()
+        videoUrl.addQueryParameter("p", element.attr("data-post"))
+        videoUrl.addQueryParameter("t", element.attr("data-type"))
+        videoUrl.addQueryParameter("n", element.attr("data-nume"))
+        videoUrl.addQueryParameter("ver", version)
+        videoUrl.addQueryParameter("rand", generateRandomString(16))
+        val videoFrame = client.newCall(GET(videoUrl.toString())).execute().body.string()
+        val embedUrl = videoFrame.substringAfter("embed_url\":\"").substringBefore("\"")
+        val referer = headers.newBuilder().add("Referer", "$baseUrl/").build()
+        val videoRegex = Regex("""m3u8|.mp4""")
+        val webViewInterceptor = client.newBuilder().addInterceptor(GetSourcesInterceptor(videoRegex)).build()
```

Comment:
Move `Regex()` and `GetSourcesInterceptor()` to a class instance to avoid repetitive initialization

Reply:
can i use by lazy or must i make a class ?

Reply:
Doing .lazy isn't that important in this case, it's doing `private val interceptor = GetSourcesInterceptor(videoRegex)` and such outside the function, so the classes aren't constantly being initialized